File: app/src/main/java/com/krenvpravo/expensenotes/ui/MainActivityContract.java
package com.krenvpravo.expensenotes.ui;

import com.krenvpravo.expensenotes.models.Transaction;

import java.util.List;

/**
 * @author Dmitry Borodin on 2017-01-19.
 */

public class MainActivityContract {
    interface View {
        void showTransactionsList(List<Transaction> transactionList);
    }
    interface Presenter {
        void onViewCreated();
        void onViewDestroyed();
        void onAddTransactionClicked();
    }
}

File: app/src/main/java/com/krenvpravo/expensenotes/ui/MainActivityContract.kt
package com.krenvpravo.expensenotes.ui

import com.krenvpravo.expensenotes.models.Transaction

/**
 * @author Dmitry Borodin on 2017-01-19.
 */

class MainActivityContract {
    internal interface View {
        fun showTransactionsList(transactionList: List<Transaction>)
    }

    internal interface Presenter {
        fun onViewCreated()
        fun onViewDestroyed()
        fun onAddTransactionClicked()
    }
}
